driver/src/orderbook/onchain_filtered_orderbook.rs

```diff
@@ -0,0 +1,44 @@
+use crate::contracts::stablex_contract::{FilteredAuctionData, StableXContract};
+use crate::models::{AccountState, Order};
+
+use super::auction_data_reader::IndexedAuctionDataReader;
+use super::StableXOrderBookReading;
+
+use anyhow::Result;
+use ethcontract::{Address, U256};
+
+pub struct OnchainFilteredOrderBookReader<'a> {
+    contract: &'a (dyn StableXContract + Sync),
+    page_size: u16,
+}
+
+impl<'a> OnchainFilteredOrderBookReader<'a> {
+    pub fn new(contract: &'a (dyn StableXContract + Sync), page_size: u16) -> Self {
```

Comment:
Should this reader also use the `OrderbookFilter` type that gets passed into the `FilteredOrderbookReader` and then if the token filter is `TokenFilter::Whitelist` also pass it to the contract?